idea/src/org/jetbrains/kotlin/idea/inspections/JavaCollectionsStaticMethodCallInspection.kt

```diff
@@ -0,0 +1,123 @@
+                                       "Java Collections static method call can be replaced with Kotlin stdlib",
+                                       ProblemHighlightType.GENERIC_ERROR_OR_WARNING,
+                                       ReplaceWithStdLibFix(methodName, firstArg.text))
+            }
+        }
+    }
+
+    private fun canReplaceWithStdLib(expression: KtDotQualifiedExpression, fqName: String, args: List<KtValueArgument>): Boolean {
+        if (!fqName.startsWith("java.util.Collections.")) return false
+        val size = args.size
+        return when (fqName) {
+            "java.util.Collections.fill" -> checkApiVersion(ApiVersion.KOTLIN_1_2, expression) && size == 2
+            "java.util.Collections.reverse" -> size == 1
+            "java.util.Collections.shuffle" -> checkApiVersion(ApiVersion.KOTLIN_1_2, expression) && (size == 1 || size == 2)
+            "java.util.Collections.sort" -> size == 1 || size == 2
+            else -> false
+        }
+    }
+
+    private fun checkApiVersion(requiredVersion: ApiVersion, expression: KtDotQualifiedExpression): Boolean {
+        val module = ModuleUtilCore.findModuleForPsiElement(expression) ?: return true
+        val version = getRuntimeLibraryVersion(module) ?: return true
+        val apiVersion = ApiVersion.parse(version.substringBefore("-")) ?: return true
+        return apiVersion >= requiredVersion
+    }
+
+}
+
+private class ReplaceWithStdLibFix(private val methodName: String, private val receiver: String) : LocalQuickFix {
+    override fun getName() = "Replace with $receiver.$methodName"
+
+    override fun getFamilyName() = name
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val expression = descriptor.psiElement as? KtDotQualifiedExpression ?: return
+        val callExpression = expression.callExpression ?: return
+        val valueArguments = callExpression.valueArguments
+        val arg1 = valueArguments.getOrNull(0)?.getArgumentExpression()
+        val arg2 = valueArguments.getOrNull(1)?.getArgumentExpression()
+        val factory = KtPsiFactory(project)
+        val newExpression = when (methodName) {
+            "fill" -> when {
+                arg1 != null && arg2 != null -> factory.createExpressionByPattern("$0.fill($1)", arg1, arg2)
+                else -> null
+            }
+            "reverse" -> when {
+                arg1 != null -> factory.createExpressionByPattern("$0.reverse()", arg1)
+                else -> null
+            }
+            "shuffle" -> when {
+                arg1 != null && arg2 != null -> factory.createExpressionByPattern("$0.shuffle($1)", arg1, arg2)
+                arg1 != null -> factory.createExpressionByPattern("$0.shuffle()", arg1)
+                else -> null
+            }
+            "sort" -> when {
+                arg1 != null && arg2 != null -> factory.createExpressionByPattern("$0.sortWith(Comparator $1)", arg1, arg2.text)
```

Comment:
This syntax works only if the argument to `Collections.sort()` is a lambda, and I don't see any check where you make sure that it is one.